kernels/python-maths/requirements.txt

```diff
@@ -0,0 +1,532 @@
+comm==0.1.3
+    # via ipykernel
+contourpy==1.0.7
+    # via matplotlib
+cycler==0.11.0
+    # via matplotlib
+debugpy==1.6.7
+    # via ipykernel
+decorator==5.1.1
+    # via ipython
+defusedxml==0.7.1
+    # via nbconvert
+dill==0.3.6
+    # via pylint
+docstring-to-markdown==0.12
+    # via python-lsp-server
+entrypoints==0.4
+    # via jupyter-client
+et-xmlfile==1.1.0
+    # via openpyxl
+executing==1.2.0
+    # via stack-data
+fastjsonschema==2.17.1
+    # via nbformat
+flake8==6.0.0
+    # via python-lsp-server
+fonttools==4.39.4
+    # via matplotlib
+fqdn==1.5.1
+    # via jsonschema
+gitdb==4.0.10
+    # via gitpython
+gitpython==3.1.31
+    # via nbdime
+h5py==3.8.0
+    # via -r kernels/python-maths/requirements.in
+idna==3.4
+    # via
+    #   anyio
+    #   jsonschema
+    #   requests
+ipyevents==2.0.1
+    # via -r kernels/python-maths/requirements.in
+ipykernel==6.23.1
+    # via
+    #   ipywidgets
+    #   metakernel
+    #   nbclassic
+    #   notebook
+    #   octave-kernel
+ipympl==0.9.3
+    # via -r kernels/python-maths/requirements.in
+ipython==8.13.2
+    # via
+    #   ipykernel
+    #   ipympl
+    #   ipywidgets
+    #   jupyterlab
+ipython-genutils==0.2.0
+    # via
+    #   ipympl
+    #   nbclassic
+    #   notebook
+ipywidgets==8.0.6
+    # via
+    #   -r kernels/python-maths/requirements.in
+    #   ipyevents
+    #   ipympl
+isoduration==20.11.0
+    # via jsonschema
+isort==5.12.0
+    # via pylint
+jedi==0.18.2
+    # via
+    #   ipython
+    #   metakernel
+    #   python-lsp-server
+jinja2==3.1.2
+    # via
+    #   jupyter-server
+    #   jupyterlab
+    #   jupyterlab-server
+    #   nbclassic
+    #   nbconvert
+    #   nbdime
+    #   notebook
+joblib==1.2.0
+    # via scikit-learn
+json5==0.9.14
+    # via jupyterlab-server
+jsonpointer==2.3
+    # via jsonschema
+jsonschema[format-nongpl]==4.17.3
+    # via
+    #   jupyter-events
+    #   jupyterlab-server
+    #   nbformat
+jupyter-client==7.4.1
+    # via
+    #   ipykernel
+    #   jupyter-server
+    #   nbclassic
+    #   nbclient
+    #   notebook
+    #   octave-kernel
+    #   voila
+jupyter-core==5.3.0
+    # via
+    #   ipykernel
+    #   jupyter-client
+    #   jupyter-server
+    #   jupyterlab
+    #   metakernel
+    #   nbclassic
+    #   nbclient
+    #   nbconvert
+    #   nbformat
+    #   notebook
+    #   voila
+jupyter-events==0.6.3
+    # via jupyter-server-fileid
+jupyter-lsp==2.1.0
+    # via jupyterlab-lsp
+jupyter-server==1.24.0
+    # via
+    #   jupyter-lsp
+    #   jupyter-server-fileid
+    #   jupyter-server-mathjax
+    #   jupyterlab
+    #   jupyterlab-git
+    #   jupyterlab-server
+    #   nbclassic
+    #   nbdime
+    #   notebook-shim
+    #   voila
+jupyter-server-fileid==0.9.0
+    # via jupyter-server-ydoc
+jupyter-server-mathjax==0.2.6
+    # via nbdime
+jupyter-server-ydoc==0.8.0
+    # via jupyterlab
+jupyter-ydoc==0.2.4
+    # via
+    #   jupyter-server-ydoc
+    #   jupyterlab
+jupyterlab==3.6.3
```

Comment:
I see there is a jump in version here. from `3.4.4` which was previously used.
This happened with all the dependencies.
Question: this might break some of the code and dependencies for users if they upgrade. Do we want to go on with these dependency upgrades?

Reply:
Good point. So I think this was the whole point raised by @colinRawlings about letting pip-tools build the `requirements.txt`. I see that in smash they user jupyterlab=3.6.1, so we be should safe here. 
I can check if I find a notebook with fancy widgets that we can use to test.

Reply:
Yes I would figure out if any code is broken. Like the widgets etc... We might want to add something to test the rendering and basic functionality with some code snippets.

Reply:
I made this [PR](https://github.com/ITISFoundation/jupyter-math/pull/20) that explain how to test widgets to a template available on Master.